network-libp2p/src/behaviour.rs

```diff
@@ -1,178 +1,178 @@
-        peer_score_params: PeerScoreParams,
+        peer_score_params: gossipsub::PeerScoreParams,
     ) -> Self {
         let public_key = config.keypair.public();
         let peer_id = public_key.to_peer_id();
 
         // DHT behaviour
         let store = MemoryStore::new(peer_id);
-        let dht = Kademlia::with_config(peer_id, store, config.kademlia);
+        let mut dht = kad::Behaviour::with_config(peer_id, store, config.kademlia);
+        // Fixme: This could be avoided with a protocol such as Autonat that properly set external addresses to the
+        // swarm and also avoids us to add addresses that are purely connection candidates.
+        dht.set_mode(Some(kad::Mode::Server));
 
         // Discovery behaviour
-        let discovery = DiscoveryBehaviour::new(
+        let discovery = discovery::Behaviour::new(
             config.discovery.clone(),
             config.keypair.clone(),
             Arc::clone(&contacts),
             clock,
         );
 
         // Gossipsub behaviour
-        let thresholds = PeerScoreThresholds::default();
-        let mut gossipsub = Gossipsub::new(MessageAuthenticity::Author(peer_id), config.gossipsub)
-            .expect("Wrong configuration");
+        let thresholds = gossipsub::PeerScoreThresholds::default();
+        let mut gossipsub = gossipsub::Behaviour::new(
+            gossipsub::MessageAuthenticity::Author(peer_id),
+            config.gossipsub,
+        )
+        .expect("Wrong configuration");
         gossipsub
             .with_peer_score(peer_score_params, thresholds)
             .expect("Valid score params and thresholds");
 
         // Identify behaviour
-        let identify_config = IdentifyConfig::new("/albatross/2.0".to_string(), public_key);
-        let identify = IdentifyBehaviour::new(identify_config);
+        let identify_config = identify::Config::new("/albatross/2.0".to_string(), public_key);
+        let identify = identify::Behaviour::new(identify_config);
 
         // Ping behaviour:
         // - Send a ping every 15 seconds and timeout at 20 seconds.
         // - The ping behaviour will close the connection if a ping timeouts.
-        let ping = PingBehaviour::new(PingConfig::new());
+        let ping = ping::Behaviour::new(ping::Config::new());
 
         // Connection pool behaviour
-        let pool = ConnectionPoolBehaviour::new(
+        let pool = connection_pool::Behaviour::new(
             Arc::clone(&contacts),
             peer_id,
             config.seeds,
             config.discovery.required_services,
         );
 
         // Request Response behaviour
-        let codec = MessageCodec::default();
-        let protocol = ReqResProtocol::Version1;
-        let config = RequestResponseConfig::default();
-        let request_response =
-            RequestResponse::new(codec, iter::once((protocol, ProtocolSupport::Full)), config);
+        let protocol = StreamProtocol::new("/nimiq/reqres/0.0.1");
```

Comment:
We could probably look into giving this a better version number at some point.

Reply:
Followed discovery protocol naming